### ISSUE TITLE
Wrong markdown rendering when showing method documentation

### ISSUE THREAD

**Issue by baco** [hrsh7th/nvim-compe#260]
When displaying documentation of a method, the signature gets interpreted as Markdown instead of being treated as a code snippet. As seen in the first screenshot `foo(*args, **kwargs)` is treated as if the intention was to make bold text using the `*` character around words:
![Screenshot from 2021-03-16 12-10-44](https://user-images.githubusercontent.com/102958/111333376-63d04a80-8651-11eb-872a-a37d91cf3f55.png)
_(screenshot using nvim-compe + python-language-server)_

The **correct display** of that method's documentation would be something like this:
![Screenshot from 2021-03-16 12-12-17](https://user-images.githubusercontent.com/102958/111333396-69c62b80-8651-11eb-8ac3-3debe509208e.png)
_(screenshot taken with CoC + jedi-language-server, for comparison)_

Also, the docstring of the method got stripped away.

The Language Server used for first Python example screenshot is `pyls`, the Language Server used in **the correct** screenshot is `jedi-language-server`.

**Update:**
Changing the Language Server, for another provider for Python, doesn't solve the issue... if not, makes it somehow worse because Markdown quotation appears on the info window now:
![Screenshot from 2021-03-16 17-35-14](https://user-images.githubusercontent.com/102958/111376386-2635e680-867e-11eb-8e57-a10c7cc6554e.png)
_(screenshot using nvim-compe + jedi-language-server)_

The Language Server used in this new screenshot is again `jedi-language-server`.

**Comment by hrsh7th** [comment]
I've updated markdown rendering. https://github.com/hrsh7th/nvim-compe/commit/8c0b843faba2a698836bee42ac793943dee91192

I have no python environment. Sorry.
Could you test with latest master?

**Comment by baco** [comment]
ok, trying out:

```shell
baco@terrapin ~/s/d/n/p/m/s/nvim-compe ((ab623db5…))> git lg HEAD~3...
* ab623db (HEAD, origin/master, origin/HEAD) Improe perf — hrsh7th (2 hours ago)
* 8c0b843 Improve markdown rendering — hrsh7th (3 hours ago)
* b5decc5 Improve a bit — hrsh7th (4 hours ago)
```

It renders more information now, like it is not truncated the hole docstring of the python method, but I still get a misinterpretation of code to render:

![image](https://user-images.githubusercontent.com/102958/111621765-2261ab80-87c7-11eb-9e4d-dacafd28d99d.png)
_(rendering with python-language-server backend)_

![image](https://user-images.githubusercontent.com/102958/111621443-c434c880-87c6-11eb-985a-5905d064187e.png)
_(rendering with jedi-language-server backend)_

**Comment by hrsh7th** [comment]
OK. I will prepare python language server.

**Comment by baco** [comment]
It is not difficult to have a python environment in userspace, if you already have python installed on a Linux.

Linux setup:
* Assume you have installed any python stable version (python3 >= 3.6) and pip (python3-pip on Debian based distros)
* Setting up python LSP on Nvim, just add the Lua setting: `lua require('lspconfig').pyls.setup{}` as for any other LServer
* Environment install:
  ```shell
  $ python3 -mvenv --clear --prompt=testing-1-2-3 $(mktemp -ud)
  $ . /tmp/tmp.MraWkvedLf/bin/activate
  ```
  and you are in a development environment where you can install packages that won't affect your system (the actual name of the temp directory may vary)
  ```shell
  (testing-1-2-3)$ python -mpip install python-language-server
  (testing-1-2-3)$ nvim /tmp/lala.py
  ```
* Then, just completing `br` word should trigger the options

**Comment by baco** [comment]
Just for adding extra information, both, `python-language-server` (pyls) and `jedi-language-server` are language providers for the python language, as well as some others like `pyright` (which I don't have installed to test because I don't have a Node environment set up). I'd be pleased if _pyls_ gets working but may be other developers that prefer any of the other options as well.

**Comment by mjlbach** [comment]
I fixed the signature markdown rendering in core https://github.com/neovim/neovim/pull/14186, not sure if it needs to be updated here

**Comment by folke** [comment]
This should be fixed as part of #426

**Comment by baco** [comment]
@folke 

> This should be fixed as part of #426

As I replied on your PR, i don't think that solves this issue
